src/lib/storage/abstract_device_factory.rb

```diff
@@ -131,12 +133,12 @@ def build_tree_recursive(parent, name, content)
           check_param(name, content.keys)
 
           # Split up pure parameters and sub-product descriptions
-          sub_prod = content.select{ |k,v|  factory_products.include?(k) }
-          param    = content.select{ |k,v| !factory_products.include?(k) }
+          sub_prod = content.select { |k, _v|  factory_products.include?(k) }
+          param    = content.select { |k, _v| !factory_products.include?(k) }
```

Comment:
not related to this pr, but when I see these two lines I immediatelly get idea using `partition` like
```
sub_prod, param = content.partition { |k, _v|  factory_products.include?(k) }
```